src/core/document.js

```diff
@@ -578,8 +578,20 @@ class Page {
     return tree;
   }
 
-  getAnnotationsData(intent) {
-    return this._parsedAnnotations.then(function (annotations) {
+  getAnnotationsData(handler, task, intent) {
+    return this._parsedAnnotations.then(annotations => {
+      const partialEvaluator = new PartialEvaluator({
+        xref: this.xref,
+        handler,
+        pageIndex: this.pageIndex,
+        idFactory: this._localIdFactory,
+        fontCache: this.fontCache,
+        builtInCMapCache: this.builtInCMapCache,
+        standardFontDataCache: this.standardFontDataCache,
+        globalImageCache: this.globalImageCache,
+        options: this.evaluatorOptions,
+      });
+      const textContentPromises = [];
```

Comment:
Please move this down to *after* the `length === 0` early return, to avoid unconditionally initializing this data.